langserver/handlers/symbol.go

```diff
@@ -0,0 +1,91 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	lsctx "github.com/hashicorp/terraform-ls/internal/context"
+	ihcl "github.com/hashicorp/terraform-ls/internal/hcl"
+	ilsp "github.com/hashicorp/terraform-ls/internal/lsp"
+	"github.com/hashicorp/terraform-ls/internal/terraform/lang"
+	"github.com/sourcegraph/go-lsp"
+)
+
+func (h *logHandler) TextDocumentSymbol(ctx context.Context, params lsp.DocumentSymbolParams) ([]lsp.SymbolInformation, error) {
+	var symbols []lsp.SymbolInformation
+
+	fs, err := lsctx.DocumentStorage(ctx)
+	if err != nil {
+		return symbols, err
+	}
+
+	pf, err := lsctx.ParserFinder(ctx)
+	if err != nil {
+		return symbols, err
+	}
+
+	file, err := fs.GetDocument(ilsp.FileHandlerFromDocumentURI(params.TextDocument.URI))
+	if err != nil {
+		return symbols, err
+	}
+
+	text, err := file.Text()
+	if err != nil {
+		return symbols, err
+	}
+
+	hclFile := ihcl.NewFile(file, text)
+
+	// TODO: block until it's available <-pf.ParserLoadingDone()
+	// requires https://github.com/hashicorp/terraform-ls/issues/8
+	// TODO: replace crude wait/timeout loop
```

Comment:
I can adjust this to max `2s`. The reason this is needed is because the `documentSymbol` request happens right away once a file is opened/changed, in my local testing the parser was never ready right away and when it errored everytime. I have never hit a max timeout either fwiw, I suspect the parser just needed a few hundred milliseconds.

Reply:
> The reason this is needed is because the documentSymbol request happens right away once a file is opened

I see, I wasn't aware of that. In that case I assume it's actually desirable to even wait longer before responding? i.e. we can and should tolerate longer timeout.

I was just applying caution because I saw some clients setting hard timeout for responses for some methods - e.g. formatting in particular is expected to be done within a second or two in some clients's implementations, otherwise the request is just cancelled. I assume the caution is not applicable to all methods then.

So feel free to keep it at 3 seconds.

Reply:
Regardless though we really need to resolve #8